ibis/pyspark/compiler.py

```diff
@@ -0,0 +1,498 @@
+
+    if how == 'sample':
+        fn = F.stddev_samp
+    elif how == 'pop':
+        fn = F.stddev_pop
+    else:
+        raise AssertionError("Unexpected how: {}".format(how))
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.Variance)
+def compile_variance(t, expr, context=None):
+    how = expr.op().how
+
+    if how == 'sample':
+        fn = F.var_samp
+    elif how == 'pop':
+        fn = F.var_pop
+    else:
+        raise AssertionError("Unexpected how: {}".format(how))
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.Arbitrary)
+def compile_arbitrary(t, expr, context=None):
+    how = expr.op().how
+
+    if how == 'first':
+        fn = functools.partial(F.first, ignorenulls=True)
+    elif how == 'last':
+        fn = functools.partial(F.last, ignorenulls=True)
+    else:
+        raise NotImplementedError
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.WindowOp)
+def compile_window_op(t, expr):
+    op = expr.op()
+    return t.translate(op.expr).over(compile_window(op.window))
+
+
+@compiles(ops.Greatest)
+def compile_greatest(t, expr):
+    op = expr.op()
+
+    src_columns = t.translate(op.arg)
+    if len(src_columns) == 1:
+        return src_columns[0]
+    else:
+        return F.greatest(*src_columns)
+
+
+@compiles(ops.Least)
+def compile_least(t, expr):
+    op = expr.op()
+
+    src_columns = t.translate(op.arg)
+    if len(src_columns) == 1:
+        return src_columns[0]
+    else:
+        return F.least(*src_columns)
+
+
+@compiles(ops.Abs)
+def compile_abs(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.abs(src_column)
+
+
+@compiles(ops.Round)
+def compile_round(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    scale = op.digits.op().value if op.digits is not None else 0
+    rounded = F.round(src_column, scale=scale)
+    if scale == 0:
+        rounded = rounded.astype('long')
+    return rounded
+
+
+@compiles(ops.Ceil)
+def compile_ceil(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.ceil(src_column)
+
+
+@compiles(ops.Floor)
+def compile_floor(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.floor(src_column)
+
+
+@compiles(ops.Exp)
+def compile_exp(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.exp(src_column)
+
+
+@compiles(ops.Sign)
+def compile_sign(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+
+    return F.when(src_column == 0, F.lit(0.0)) \
+        .otherwise(F.when(src_column > 0, F.lit(1.0)).otherwise(-1.0))
+
+
+@compiles(ops.Sqrt)
+def compile_sqrt(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.sqrt(src_column)
+
+
+@compiles(ops.Log)
+def compile_log(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.log(float(op.base.op().value), src_column)
```

Comment:
You probably want to translate the first argument here.

Reply:
Fixed